op-chain-ops/genesis/migration_action/action.go

```diff
@@ -0,0 +1,110 @@
+package migration_action
+
+import (
+	"context"
+	"math/big"
+	"path/filepath"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/hardhat"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis/migration"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type Config struct {
+	DeployConfig          *genesis.DeployConfig
+	OVMAddressesPath      string
+	EVMAddressesPath      string
+	OVMAllowancesPath     string
+	OVMMessagesPath       string
+	EVMMessagesPath       string
+	Network               string
+	HardhatDeployments    []string
+	L1URL                 string
+	StartingL1BlockNumber uint64
+	L2DBPath              string
+	DryRun                bool
+}
+
+func Migrate(cfg *Config) (*genesis.MigrationResult, error) {
```

Comment:
This is identical to what we had in CLI, but extracted into a standalone method so that the migration can call it directly.